MiniDB/DBTransaction.cs

```diff
@@ -0,0 +1,145 @@
+    ///   TODO: add resolve transaction - cannot be undone
+    /// </summary>
+    public enum TransactionType
+    {
+        /// <summary>
+        /// An unkown transaction type (default)
+        /// </summary>
+        Unknown = -1,
+        
+        /// <summary>
+        /// Adding an item
+        /// </summary>
+        Add,
+        
+        /// <summary>
+        /// Removing an item
+        /// </summary>
+        Delete,
+        
+        /// <summary>
+        /// Changing an item
+        /// </summary>
+        Modify,
+        
+        /// <summary>
+        /// An undo change
+        /// </summary>
+        Undo,
+
+        /// <summary>
+        /// An redo change
+        /// </summary>
+        Redo,
+    }
+
+    /// <summary>
+    /// Transaction class to save information about changes to the Database of type T.
+    /// </summary>
+    /// <typeparam name="T">DatabaseObject type that is stored in DB</typeparam>
+    public class DBTransaction<T> : DatabaseObject where T : DatabaseObject
+    {
+        /// <summary>
+        /// Key to use when removing items
+        /// </summary>
+        public const string ItemRemovedConstKey = "ITEM_REMOVED";
+
+        /// <summary>
+        /// Key to use on added items
+        /// </summary>
+        public const string ItemAddConstKey = "ITEM_ADDED";
+
+        /// <summary>
+        /// Keep track of the next available transaction ID
+        /// </summary>
+        private static int nextAvailableID = 0;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBTransaction{T}" /> class.
+        /// </summary>
+        public DBTransaction() : base()
+        {
+            this.ID = nextAvailableID;
+            this.Transaction_timestamp = System.DateTime.Now;
+            this.Active = true;
+            nextAvailableID++;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBTransaction{T}" /> class.
+        ///   Copy constructor by value
+        /// </summary>
+        /// <param name="inTransaction">The transaction to copy</param>
+        internal DBTransaction(DBTransaction<T> inTransaction) : base()
+        {
+            this.ID = inTransaction.ID;
+            this.Transaction_timestamp = inTransaction.Transaction_timestamp;
+            this.Transacted_item = inTransaction.Transacted_item;
+            this.TransactionType = inTransaction.TransactionType;
+            this.Item_ID = inTransaction.Item_ID;
+            this.Changed_property = inTransaction.Changed_property;
+            this.Property_new = inTransaction.Property_new;
+            this.Property_old = inTransaction.Property_old;
+            
+            // needed? - keep until collision detecting is implemented
+            nextAvailableID++;
+        }
+
+        #region transacted object properties
+        /// <summary>
+        /// Gets or sets the ID of the item that was acted on
```

Comment:
these code comments are useless. Either make them say "why" instead of "what" or just delete them. And if they're there because Visual Studio wants them there, then you might as well remove them, because then you at least get a warning about having no documentation instead of the false security of bad documentation.